### PR TITLE
Export inference test data and add FUNC_TIME_MEASUREMENT macro for test & debug

### PR DESCRIPTION
**Pull request by ruimashita**
Export inference_test_data directory that include input sample and expected results as npy files from tensorflow for inference test.
And add FUNC_TIME_MEASUREMENT macro when make elf file for debug.

Solve #183

## Description

* Export input sample and expected results as npy files for each oeperations from tensorflow.
* add `FUNC_TIME_MEASUREMENT` when make debug `elf`
* Delete unused `__WITHOUT_TEST__` macro

## How has this been tested?

After run this convert command.
```
$ ./blueoil.sh convert ~/blueoil/example/det.yml ~/blueoil/saved/det_20190326181434/
```

Found input sample and expected results as npy files for each oeperations.
```
   ls saved/det_20190326181434/export/save.ckpt-161/128x128/inference_test_data/
000_images_placeholder:0.npy                                              171_block_6_1x1_bn_beta_read:0.npy
001_block_1_conv_kernel:0.npy                                             172_block_6_1x1_bn_gamma:0.npy
....
...
```

I can measure time when run elf with input and expected result. 
```
$ ./models/lib/lm_fpga.elf ../inference_test_data/000_images_placeholder\:0.npy ../inference_test_data/337_output\:0.npy
-------------------------------------------------------------
Comparison: Default network test  succeeded!!!
-------------------------------------------------------------
BatchNorm,5460,142,  sum:5.602ms
Convolution,6218,697,  sum:6.915ms
ExtractImagePatches,484,154,106,38,22,9,8,8,  sum:0.829ms
Maximum,48,  sum:0.048ms
Mul,48,  sum:0.048ms
Packing input for kn2row,2246,408,193,88,53,29,14,36,56,119,49,19,14,18,14,  sum:3.356ms
QConv2D kn2row tiling,7611,2508,1940,1425,2362,3754,2961,190,139,192,13388,2238,2921,2229,2773,  sum:46.631ms
QTZ_linear_mid_tread_half,6724,  sum:6.724ms
QuantizedConv2D,2901,  sum:2.901ms
QuantizedConv2D_ApplyScalingFactor,107,  sum:0.107ms
TotalInitTime,22054,  sum:22.054ms
TotalRunTime,79626,  sum:79.626ms
func_ConcatOnDepth,43,  sum:0.043ms
kn2row-1x1,6207,691,  sum:6.898ms
matrix_multiplication,6194,682,  sum:6.876ms
matrix_transpose (row_major),129,  sum:0.129ms
pack_input_to_qwords,2226,398,184,81,43,22,7,28,46,112,42,11,7,10,7,  sum:3.224ms
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ruimashita** [comment]
@nlpng @lm-jira @tkclimb 
Plz review 👍

**Comment by iizukak** [comment]
@nlpng @lm-jira Review please.

**Comment by tkng** [comment]
I don't prefer the name 'npy', because it's just a abbreviation of `numpy`, IMO it not adequate to describe "expected result for inrefence test".

**Comment by ruimashita** [comment]
@tkng 
Sure 👍  Thanks. I will rewrite description of this PR and git merge log.

**Comment by tkng** [comment]
Oops, sorry... :bowing_man: What I want to argue were:

- at least we should change the directory name 'npy' before unveiling the feature to public.
- according to that, it would be better to change some documents/code. e.g. in `lmnet/executor/export.py`, we can find some variable names like `npy_output_dir`, but I think it is not understandable to beginners.

Since it is hard to change the interface later, it would be better to clean up the interface before going to public.

**Comment by tkng** [comment]
export doesn't use the directory name 'npy', please forget the comment above.

**Comment by ruimashita** [comment]
@tkng 
Ohhh, I had misunderstood 😿 Then I agree what you say 👍 

> according to that, it would be better to change some documents/code. e.g. in lmnet/executor/export.py, we can find some variable names like npy_output_dir, but I think it is not understandable to beginners.

Just now, It is the time to rename variables, I will do it ❗️❗️❗️ 
 (I will not do forever if I miss this opportunity 🚬 )

**Comment by ruimashita** [comment]
@nlpng @lm-jira 
Currently, The directory name for inference test is `./saved/{experiment id}/export/{checkpoint name}/{image size}/{image name}`.
i.e. : 
```
./saved/cls_20190226171345/export/save.ckpt-27/128x128/cat.jpg/
├──000_images_placeholder:0.npy 
├──....
```

`{image name}` part of this dir name is not clearly for usage.
I want to rename this part to `inference_test_sample`,
i.e.
```
./saved/cls_20190226171345/export/save.ckpt-27/128x128/inference_test_sample/
├──000_images_placeholder:0.npy 
├──....
```
How do you think about this `inference_test_sample` dir name?

**Comment by lm-jira** [comment]
> How do you think about this `inference_test_sample` dir name?

I think the dir name should have the word `operations` because it contains the list of operations.
What's about `test_inference_operations`?

**Comment by ruimashita** [comment]
@lm-jira 
Thanks. The name of the `test_inference_operations` looks to contain test scripts for me.
And the dir contains operations outputs and  inputs `raw_image.npy`, `preprocessed_image.npy`, moreover `000_images_placeholder:0.npy` is `tf.placeholder` operation output But this is input of network and inference test. The `sample` means feeding input and expected output of inference test, but it looks not good name. 
More simply, how is `inference_test_data` or `inference_test_fixtures` ?

**Comment by lm-jira** [comment]
@ruimashita You're right. The name should not start with `test`.

> More simply, how is `inference_test_data` or `inference_test_fixtures` ?

I prefer to use `inference_test_data` because for me it's more than a test fixture.

**Comment by ruimashita** [comment]
@iizukak @lm-jira @nlpng 
I update source code and description of PR.
Plz review again 👍

**Comment by ruimashita** [comment]
run lmnet test

**Comment by ruimashita** [comment]
run lmnet test

File: lmnet/executor/export.py [hunk 1]
```diff
@@ -44,6 +50,7 @@ def _pre_process(raw_image, pre_processor, data_format):
 
 
 def _save_npy(image_path, output_dir, image, raw_image, all_outputs, image_size):
+    shutil.copy(image_path, os.path.join(output_dir))
     shutil.copy(image_path, os.path.join(output_dir, "raw_image.png"))
```
Comment (nlpng) [inline lmnet/executor/export.py]: Do we need to copy same image to different name? Maybe I missed something.

Comment (ruimashita) [reply lmnet/executor/export.py]: The line `shutil.copy(image_path, os.path.join(output_dir))` copy input image to `out_dir` with reserving input image file name.
The line `shutil.copy(image_path, os.path.join(output_dir, "raw_image.png"))` copy input image to `out_dir` with changing image file name `raw_image.png`.
We need one of them, But I don't know which one should be choose for useful to debug.

Comment (nlpng) [reply lmnet/executor/export.py]: Thanks, I see, so keeping both is better for debugging purpose, feel that only the `raw_image.png` should be ok though.

**Comment by lm-jira** [comment]
LGTM too

**Pull request merged**